File: PercusionaTec-main/Assets/Scripts/PHP/registrarDatos.php
<?php

// Autores: Jose Luis Madrigal, Cesar Emiliano Palome, Jose Angel Garcia y Erika Marlene
// Codigo para obtener datos de forma y mandarlos con un query a la BD (con valor de hora inicio)
include 'conexion.php';
    if ($_POST) {
        $usuario = $_POST["usuario"];
        $nombre = $_POST["nombre"];
        $ciudad = $_POST["ciudad"];
        $mail = $_POST["mail"];
        $contrasena = $_POST["contrasena"];
        $fechaNaci = $_POST["fechaNaci"];
        $nacionalidad = $_POST["nacionalidad"];
        
        $sql_agregar1 = "INSERT INTO `jugador` (`Id_usuario`, `nombre`, `ciudad`, `pais`, `email`, `contraseña`, `fecha_nacimiento`, `pais`) VALUES (?, ?, ?, ?, ?, ?, ?);";
        $sentencia_agregar1 = $pdo->prepare($sql_agregar1);
        $resultado1 = $sentencia_agregar1->execute(array($usuario,$nombre,$ciudad,$mail,$contrasena,$fechaNaci,$nacionalidad));
        if ($resultado1==true) {
            $sentencia_agregar1 = null;
            $pdo = null;
            echo "User: " . $usuario . "\n";
            echo "Name: " . $nombre . "\n";
            echo "City: " . $ciudad . "\n";
            echo "Mail: " . $mail . "\n";
            echo "Password: " . $contrasena . "\n";
            echo "Date of birth: " . $fechaNaci . "\n";
            echo "Nationality: " . $nacionalidad . "\n";
            } else {
            echo "Error al insertar en la BD: ";
            echo $pdo->errorInfo();
            }
            
        
        
    } else {
        echo "No hay método POST";
    }

?>

File: PercusionaTec-main/Assets/Scripts/RegistrarDatos.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;
using TMPro;
using UnityEngine.Networking;
using UnityEngine.SceneManagement;
// Autores: Jose Luis Madrigal, Erika Marlene Garcia, Cesar Emiliano Palome, Jose Angel Garcia
// Clase para hacer conexion web con unity para registrar datos ingresados

public class RegistrarDatos : MonoBehaviour
{ 
        
    public TextMeshProUGUI resultado;

    //Los datos de entrada
    public TMP_InputField textoUsuario;
    public TMP_InputField textoNombre;
    public TMP_InputField textoCiudad;
    public TMP_InputField textoMail;
    public TMP_InputField textoContrasena;
    public TMP_InputField textoFechaNaci;
    public TMP_InputField textoNacionalidad;

    
    //Enviar los datos al servidor(click del boton)
    public void EnviarDatos()
{
    StartCoroutine(SubirDatos());
}

    private IEnumerator SubirDatos()
    {
    //Recuperar los datos
    string usuario = textoUsuario.text;
    string nombre = textoNombre.text;
    string ciudad = textoCiudad.text;
    string mail = textoMail.text;
    string contrasena = textoContrasena.text;
    string fechaNaci = textoFechaNaci.text;
    string nacionalidad = textoNacionalidad.text;
    //Crear un objeto con los datos
    WWWForm forma = new WWWForm();
    forma.AddField("usuario", usuario);
    forma.AddField("nombre", nombre);
    forma.AddField("ciudad", ciudad);
    forma.AddField("mail", mail);
    forma.AddField("contrasena", contrasena);
    forma.AddField("fechaNaci", fechaNaci);
    forma.AddField("nacionalidad", nacionalidad);

    string URLverificaJugador = "https://percusionatec-ge4wk.ondigitalocean.app/jugador/" + usuario;
    UnityWebRequest requestVerif = UnityWebRequest.Get(URLverificaJugador);
    yield return requestVerif.SendWebRequest();
    //....despues de cierto tiempo
    if (!(requestVerif.result == UnityWebRequest.Result.Success))
    {
        string URLregistroJugador = "https://percusionatec-ge4wk.ondigitalocean.app/jugador/";
        UnityWebRequest requestRegistro = UnityWebRequest.Post(URLregistroJugador, forma);
        yield return requestRegistro.SendWebRequest();
        if (requestRegistro.result == UnityWebRequest.Result.Success)
        {
            string texto = requestRegistro.downloadHandler.text;
            resultado.text = texto;
        }
        else
        {
            resultado.text = "Error al ingresar datos";
        }
    }
    else
    {
        resultado.text = "Error, el usuario ya está registrado ";
    }
    }  

}
